### PR TITLE
Ditch setup.cfg and replace it with pyproject.toml

### PR DESCRIPTION
**Pull request by santisoler**
Remove the `setup.cfg` file, and use `pyproject.toml` to package Choclo. Move flake8 configuration to its own `.flake8` file.

**Relevant issues/PRs:**

Related to https://github.com/fatiando/community/issues/143

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,50 @@
...
+]
+
+[project.urls]
+"Homepage" = "https://www.fatiando.org/choclo"
+"Documentation" = "https://www.fatiando.org/choclo"
+"Repository" = "https://github.com/fatiando/choclo.git"
+"Changelog" = "https://www.fatiando.org/choclo/latest/changes.html"
+"Bug Tracker" = "https://github.com/fatiando/choclo/issues"
+"Source Code" = "https://github.com/fatiando/choclo"
+"Release Notes" = "https://github.com/fatiando/choclo/releases"
```
[the remark this answers is not in the document]
Comment (santisoler) [reply pyproject.toml:44]: It's probably overkill. I don't remember where I got those possible fields from, but checking out [the official documentation](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/), having Homepage, Documentation, Repository, Issues and Changelog should be enough.

[the remark this answers is not in the document]
Comment (santisoler) [reply pyproject.toml:44]: Now that I see the `pyproject.toml` we have in other packages, maybe I merged a few of those with the ones I found in the documentation:

https://github.com/fatiando/pooch/blob/78945b761bf132af58c7437c85e055c4225df84d/pyproject.toml#L42-L46

[the remark this answers is not in the document]
Comment (santisoler) [reply pyproject.toml:44]: I think we can decide on a set of urls every project might need and set them on every project. For example, Pooch in PyPI is currently showing the Homepage, and that will go away if we make a new release:

image

[the remark this answers is not in the document]
Comment (leouieda) [reply pyproject.toml:44]: Agreed! I don't think we need homepage since it's the same as documentation.

**Review by leouieda — APPROVED** [review]

**Pull request merged**